### PR TITLE
Refactor test_generation.py: Parameterize text model tests and add edge case

### PR DESCRIPTION
**Pull request by borgesenioc**
- Used @pytest.mark.parametrize to simplify and reuse dialog tests in TestTextModelInference.
- Added a new test case for handling empty dialog inputs to improve edge case coverage.
- Maintained the vision model tests in skipped state with @unittest.skip and @pytest.mark.skip.
- Improved overall readability and modularity of the test file.

### CODE REVIEW

File: models/llama3/tests/api/test_generation.py [hunk 1]
```diff
@@ -42,33 +39,44 @@ class TestTextModelInference(unittest.TestCase):
     def setUpClass(cls):
         cls.generator = build_generator("TEXT_MODEL_CHECKPOINT_DIR")
 
-    def test_run_generation(self):
-        dialogs = [
+    @pytest.mark.parametrize(
+        "dialogs",
```
Comment (ashwinb) [inline models/llama3/tests/api/test_generation.py:43]: shouldn't the parameter be `dialog` and the test doesn't loop at all?

**Review by ashwinb — CHANGES_REQUESTED** [review]
